packages/devtools_app/lib/src/provider/provider_list.dart

```diff
@@ -0,0 +1,213 @@
+
+    if (controller.state == null) {
+      if (ids.isNotEmpty) controller.state = ids.first;
+      return;
+    }
+
+    if (ids.isEmpty) {
+      controller.state = null;
+    } else if (!ids.contains(controller.state)) {
+      controller.state = ids.first;
+    }
+  }, onError: (err) {
+    // nothing to do here, but passing onError prevents tests from failing when
+    // testing scenarios where providerIdsStream emits an error
+  });
+
+  ref.onDispose(sub.cancel);
+
+  return null;
+});
+
+class ProviderList extends StatefulWidget {
+  const ProviderList({Key key}) : super(key: key);
+
+  @override
+  _ProviderListState createState() => _ProviderListState();
+}
+
+class _ProviderListState extends State<ProviderList> {
+  final scrollController = ScrollController();
+
+  @override
+  void dispose() {
+    scrollController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Consumer(
+      builder: (context, watch, child) {
+        final state = watch(providerIdsProvider);
+
+        return state.when(
+          loading: () => const Center(child: CircularProgressIndicator()),
+          error: (err, stack) => const Padding(
+            padding: _tilePadding,
+            child: Text('<unknown error>'),
+          ),
+          data: (providerNodes) {
+            return Scrollbar(
+              controller: scrollController,
+              isAlwaysShown: true,
+              child: ListView.builder(
+                primary: false,
+                controller: scrollController,
+                itemCount: providerNodes.length,
+                itemBuilder: (context, index) {
+                  return ProviderScope(
+                    key: Key('provider-${providerNodes[index]}'),
+                    overrides: [
+                      _providerIdProvider
+                          .overrideWithValue(providerNodes[index])
+                    ],
+                    child: const ProviderNodeItem(),
+                  );
+                },
+              ),
+            );
+          },
+        );
+      },
+    );
+  }
+}
+
+class ProviderNodeItem extends ConsumerWidget {
+  const ProviderNodeItem({Key key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context, ScopedReader watch) {
+    final providerId = watch(_providerIdProvider);
+    final state = watch(providerNodeProvider(providerId));
+
+    final isSelected = watch(_isSelectedProvider);
+
+    final colorScheme = Theme.of(context).colorScheme;
+    final backgroundColor =
+        isSelected ? colorScheme.selectedRowBackgroundColor : null;
+
+    return GestureDetector(
+      behavior: HitTestBehavior.opaque,
+      onTap: () => context.read(selectedProviderIdProvider).state = providerId,
+      child: Container(
+        color: backgroundColor,
+        padding: _tilePadding,
+        child: state.when(
+          loading: () => const CenteredCircularProgressIndicator(),
```

Comment:
can we get rid of these spinners? It looks like there is a single spinner for the whole container, and then there is a spinner for each row. It created a kind of strange multi-spinner effect that we probably don't want:
![Screen Shot 2021-04-14 at 12 52 31 PM](https://user-images.githubusercontent.com/43759233/114776585-7e1c4780-9d27-11eb-82c9-1d80751df50f.png)

Reply:
Should be good now